bin/start-systemtest

```diff
@@ -0,0 +1,247 @@
+  local TASK=${1}
+  local DATASET=$(echo ${TASK} | jq '.dataset' | tr -d \" | tr '[:upper:]' '[:lower:]')
+
+  echo "${DATASET}"
+}
+
+get_state(){
+  # This function receives information from a SAPS task and obtains the processing status it is in.
+  # There is a flow in which a task follows, which is:
+  # created -> downloading -> downloaded -> preprocessing -> ready -> running -> finished -> archiving -> archived
+  # In any of the transitions errors can occur and go to the 'failed' state
+  # The output of this function is formatted for lower case and the initial and final quotation marks are removed.
+  local	TASK=${1}
+  local STATE=$(echo ${TASK} | jq '.state' | tr -d \" | tr '[:upper:]' '[:lower:]')
+
+  echo "${STATE}"
+}
+
+assert(){
+  local ID=${1}
+  local TEST_CASE=${2}
+  local TASK=$(get_task_by_id ${ID})
+  local STATE=$(get_state "${TASK}")
+  local DATASET=$(get_dataset "${TASK}")
+
+  echo "($(get_timestamp)) [${TEST_CASE}] Checking task situation [${ID}]..."
+
+  if [[ ${TEST_CASE} == 'endtoend-test' && ${DATASET} == 'landsat_7' ]]; then
+    echo "($(get_timestamp)) [${TEST_CASE}] Test failed as expected."
+  elif [[ ${STATE} == 'archived' ]]; then
+    echo "($(get_timestamp)) [${TEST_CASE}] Successful test."
+    check_files ${ID} ${TEST_CASE}
+  else
+    echo "($(get_timestamp)) [${TEST_CASE}] Test failure."
+    exit 1
+  fi
+}
+
+check_files(){
+  local ID=${1}
+  local TEST_CASE=${2}
+
+  if [[ ${TEST_CASE} == 'endtoend-test' ]]; then
+    check_md5sum_files ${ID} ${TEST_CASE}
+  else
+    echo "($(get_timestamp)) [${TEST_CASE}] No need to check files in ${TEST_VERSION}."
+  fi
+}
+
+check_tasks(){
+  local TASK_IDS=${1}
+  local TEST_CASE=${2}
+  local TIMESLEEP=$(get_timesleep ${TEST_CASE})
+
+  echo "($(get_timestamp)) [${TEST_CASE}] Checking submitted tasks..."
+
+  # For each id of submitted tasks in this ${TEST_CASE}
+  #   While the task is processing
+  #     Get new task information by id and obtain the state
+  #     If the state is terminal ('archived' or 'failed'), then
+  #       Assert that the terminal state for the case of this task is as expected and skip to the next iteration
+  #     Else,
+  #       Sleep for X seconds (X is based on ${TEST_CASE}, to decrease access to the dispatcher's backend because some tests take longer than others)
+  for ID in $(echo "${TASK_IDS}" | jq ".[]" | tr -d \")
+  do
+    while true; do
+       local TASK=$(get_task_by_id ${ID})
+       local STATE=$(get_state "${TASK}")
+
+       echo "($(get_timestamp)) [${TEST_CASE}] Current task state [${ID}]: ${STATE}"       
+       if [ ${STATE} == 'archived' -o ${STATE} == 'failed' ]; then
+         echo "($(get_timestamp)) [${TEST_CASE}] Task [${ID}] has finished processing."
+         assert ${ID} ${TEST_CASE}
+         break
+       fi
+       echo "($(get_timestamp)) [${TEST_CASE}] Waiting ${TIMESLEEP} seconds to check again..."
+       sleep ${TIMESLEEP}
+    done
+  done
+}
+
+submit_task(){
```

Comment:
it would be nice to doc here the format of the output (i guess it is the task_ids)